doc/development.md

```diff
@@ -89,4 +89,205 @@
+using a hanging indent. When using a hanging indent, the following should
+be considered:
+- There should be no arguments on the first line.
+- Further indentation should be used to clearly distinguish itself as a continuation line.
+- Hanging indentation should stick to the 4-space rule.
+```python
+# Correct
+# Add 4 spaces (an extra level of indentation) to distinguish arguments from the rest.
+foo = long_function_name(
+        var_one, var_two, var_three, var_four)
+
+# Also with if-statements.
+if (condition_1 and
+        condition_2):
+    do_something()
+
+# One line per argument for cases where arguments expand to more than one line.
+def long_function_name(
+        var_one,
+        var_two,
+        var_three,
+        var_four):
+    print(var_one)
+
+# Hanging indents should add a level.
+foo = long_function_name(
+    var_one, var_two, var_three, var_four)
+```
+```python
+# Wrong:
+# Arguments on first line forbidden (even when vertically aligned).
+foo = long_function_name(var_one, var_two,
+                         var_three, var_four)
+
+# Further indentation required as indentation is not distinguishable.
+def long_function_name(
+    var_one, var_two, var_three, var_four):
+    print(var_one)
+
+```
+
+To break comprehension Lists, Dicts, Sets as well as generator expressions,
+keep expressions that would be written in a single line together.
+Continuation lines stick to the same rules as above. Closing
+brace/bracket/parenthesis shall line up under the first character of the
+line start.
+
+```python
+# Correct
+# Vertically aligned and extra 4 spaces for condition continuation
+foo = [var for var in bar_list
+       if condition_1 and
+           condition_2]
+
+# Closing bracket starts under the first character
+foo = [
+    var for var in bar_list if condition_1
+]
+```
+
+```python
+# Wrong
+# for-statement broken in multiple lines
+foo = [var for var
+       in bar_list if condition_1 and
+       condition_2]
+```
+
+### Docstrings
+
+Always use the three double-quote `"""` format for docstrings 
+(per [PEP 257](https://peps.python.org/pep-0257/)).
+A docstring should be organized as multiple summary lines (one physical line
+not exceeding 79 characters) terminated by a period. First and last line
+of the docstring shall be blank.
+
+Functions and classes must have a docstring, unless:
+- Not externally visible and very short.
+- Pertains to a test module.
+
+The docstring should be imperative-style (e.g., `Verify configuration file.`)
+rather than descriptive-style (e.g., `Verifies configuration file.`).
+The docstring should describe the function's main purpose, its calling syntax,
+and/or semantics, but avoid implementation details.
+
+Certain aspects of a function should be documented in special sections,
+listed below.
+
+#### Args: (or Arguments:)
+
+List each parameter by name. The type of the parameter may optionally follow
+the name, enclosed in parentheses, and then the description of the parameter
```

Comment:
Providing type information for all the parameters and return values should be strongly encouraged.  It can take several forms: the type name in parentheses, clear wording in the argument description, or type annotation.